### PR TITLE
Data Storage System to Compute Engine API

### PR DESCRIPTION
**Pull request by kyliedelamater**

### CODE REVIEW

File: src/api/DataStorageSystem.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+public interface DataStorageSystem {
+
+	ReadInputResponse readInput(ReadInputRequest readInputRequest);
+	
+	WriteInputResponse writeInput(WriteInputRequest writeInputRequest);
+	
+	ReadOutputResponse readOutput(ReadOutputRequest readOutputRequest);
+	
+	WriteOutputResponse writeOutput(WriteOutputRequest writeOutputRequest);
+}
```
Comment (EDOLK) [inline src/api/DataStorageSystem.java:8]: Its unlikely that the data storage system will ever need to read from output or write to input, I'd recommend removing the methods `readOutput` and `writeInput`, and the relevant request/response interfaces.

File: src/api/PrototypeDataApi.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+		//write to user-specified input
+		WriteInputResponse writeInputResponse = new WriteInputRequest();
+		
+		//read from user-specified output
+		ReadOutputResponse readOutResponse = new ReadOutputRequest();
+		
+		//write to user-specified output
+		WriteOutputResponse writeOutputResponse = new WriteOutputRequest();
+		
+	}
```
Comment (EDOLK) [inline src/api/PrototypeDataApi.java:13]: This method should be sending requests to and getting responses from our data storage system, something like:
`ReadInputResponse readInputResponse = data.readInput(new ReadInputRequest());`
and:
`WriteOutputResponse writeOutputResponse = data.writeOutput(new WriteOutputRequest());`

File: src/api/ReadInputRequest.java [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package api;
+
+public class ReadInputRequest implements ReadInputResponse {
```
Comment (EDOLK) [inline src/api/ReadInputRequest.java]: ReadInputRequest is not a type of ReadInputResponse, and shouldn't be implementing this interface.

File: src/api/WriteOutputRequest.java [hunk 4]
```diff
@@ -0,0 +1,5 @@
+package api;
+
+public class WriteOutputRequest implements WriteOutputResponse {
```
Comment (EDOLK) [inline src/api/WriteOutputRequest.java]: WriteOutputRequest is not a type of WriteOutputResponse, and shouldn't be implementing this interface.

**Review by EDOLK — APPROVED** [review]
You definitely need to change your prototype class before merging, the methods in DataStorageSystem and the interfaces are less important but I would consider changing them.

File: src/api/PrototypeDataApi.java (hunk 2 above)
Comment (kyliedelamater) [reply src/api/PrototypeDataApi.java:13]: thank you, this makes way more sense!

**Pull request merged**